go.mod

```diff
@@ -38,14 +38,14 @@ require (
 	golang.org/x/mod v0.9.0
 	golang.org/x/text v0.8.0
 	helm.sh/helm/v3 v3.10.3
-	k8s.io/api v0.26.1
-	k8s.io/apimachinery v0.26.1
-	k8s.io/client-go v0.26.1
-	k8s.io/component-base v0.26.1
+	k8s.io/api v0.26.2
```

Comment:
This puts us slightly ahead of CAPI, which is still at v0.26.1 for these k8s.io packages. IDK if we have a policy around that. In practice it's probably ok since it's just a patch release bump, but we should see what others think.

Reply:
I almost didn't realize that we were bumping to _`0.26.2`_. Thanks for pointing that out! Umm, we can put this PR back onto hold or leave it as it. I am fine with either! 
In case we leave it as is, its better we remove `ok-to-test` label to stop this PR from rebasing->testing every time `main` gets updated I guess.

Reply:
I don't think we need to worry about this in the abstract (especially for patch releases, but I think in general).

If there is some shared downstream k8s.io library dependency between CAPI and CAPZ in CAPZ's compile + link flow (sorry using old-fashioned terminology here) that isn't mutually compatible, then go.mod should in almost all cases complain about that, and in a scenario in which it somehow doesn't (I am being very hypothetical here) then certainly any breaking changes in forward-moving k8s.io libraries that aren't present in our declared dependent version of CAPI will break in CI in one ugly way or another.

Does that make sense? tl;dr I don't think we have to worry about this, our CI will worry about it for us.

cc @CecileRobertMichon

Reply:
Shall we remove the hold?

Reply:
> Shall we remove the hold?

I think so. It ought to be safe and it makes sense to keep up with cloud-provider-azure.